flowcraft/templates/run_mlst.py

```diff
@@ -0,0 +1,431 @@
+    Main executor of the mlst template.
+
+    Parameters
+    ----------
+    sample_id : str
+        Sample Identification string.
+    assembly : str
+        Fasta file.
+    expected_species : str
+        Expected species
+
+    """
+
+    pass_qc = False
+
+    novel_alleles = '{}_mlst_novel_alleles.fasta'.format(sample_id)
+
+    cli = [
+        'mlst',
+        '--novel',
+        novel_alleles,
+        assembly
+    ]
+
+    logger.debug("Running mlst subprocess with command: {}".format(cli))
+
+    p = subprocess.Popen(cli, stdout=PIPE, stderr=PIPE)
+    stdout, stderr = p.communicate()
+
+    # Attempt to decode STDERR output from bytes. If unsuccessful, coerce to
+    # string
+    try:
+        stderr = stderr.decode("utf8")
+        stdout = stdout.decode("utf8")
+    except (UnicodeDecodeError, AttributeError):
+        stderr = str(stderr)
+        stdout = str(stdout)
+
+    logger.info("Finished mlst subprocess with STDOUT:\\n"
+                "======================================\\n{}".format(stdout))
+    logger.info("Fished mlst subprocess with STDERR:\\n"
+                "======================================\\n{}".format(stderr))
+    logger.info("Finished mlst with return code: {}".format(
+        p.returncode))
+
+    with open(".status", "w") as fh:
+        if p.returncode != 0:
+            fh.write("error")
+            return
+        else:
+            fh.write("pass")
+
+    if p.returncode == 0:
+
+        with open('{}.mlst.txt'.format(sample_id), 'wt') as writer:
+            writer.write(stdout)
+
+        # str
+        scheme_mlst = stdout.splitlines()[0].split('\\t')[1].split('_')[0]
+        # str
+        st = stdout.splitlines()[0].split('\\t')[2]
+        # list
+        profile = stdout.splitlines()[0].split('\\t')[3:]
+
+        # In case it's an unkown ST, cleans the novel alleles file.
+        if st == '-':
+            clean_novel_alleles(novel_alleles=novel_alleles, scheme_mlst=scheme_mlst, profile=profile)
+        else:
+            # in case mlst fails to create the novel alleles file
+            if os.path.isfile(novel_alleles):
+                os.remove(novel_alleles)
+
+        if not expected_species == "PASS":
+
+            # returns the schema for the expected species, the genus for that species and the genus for the mlst schema
+            scheme, species_genus, mlst_scheme_genus = getScheme(expected_species)
+
+            if scheme_mlst.split('_', 1)[0] == scheme.split('_', 1)[0]:
+                pass_qc = True
+            else:
+                if scheme == 'unknown' and scheme_mlst != '-':
+                    pass_qc = True
+                    logger.warning("Found {scheme_mlst} scheme for a species with unknown"
+                                   " scheme".format(scheme_mlst=scheme_mlst))
+
+                elif scheme == 'unknown' and scheme_mlst == '-':
+                    pass_qc = True
+
+                # a special warning was requested for yersinia
```

Comment:
requested by whom?

Reply:
That doesn't make sense to be in documentation. Just tell what the code does and not who requested what.